internal/serve/httphandler/mfa_handler.go

```diff
@@ -81,11 +67,29 @@ func (h MFAHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	userID, err := h.AuthManager.GetUserID(ctx, token)
+	user, err := h.AuthManager.GetUser(ctx, token)
 	if err != nil {
 		httperror.InternalError(ctx, "Cannot get user ID", err, nil).Render(rw)
 		return
 	}
-	log.Ctx(ctx).Infof("[UserLogin] - Logged in user with account ID %s", userID)
+
+	userRoleCanBypassReCAPTCHA := slices.Contains(user.Roles, data.APIUserRole.String())
+
+	// validating reCAPTCHA Token
+	if !h.ReCAPTCHADisabled && !userRoleCanBypassReCAPTCHA {
+		isValid, recaptchaErr := h.ReCAPTCHAValidator.IsTokenValid(ctx, reqBody.ReCAPTCHAToken)
+		if recaptchaErr != nil {
+			httperror.InternalError(ctx, "Cannot validate reCAPTCHA token", recaptchaErr, nil).Render(rw)
+			return
+		}
+
+		if !isValid {
+			log.Ctx(ctx).Errorf("reCAPTCHA token is invalid for request with email")
+			httperror.BadRequest("reCAPTCHA token invalid", nil, nil).Render(rw)
+			return
+		}
+	}
+
+	log.Ctx(ctx).Infof("[UserLogin] - Logged in user with account ID %s", user.ID)
```

Comment:
🚨 Same here. By moving the recaptcha part past the authentication calls, you;re defeating the purpose of the recaptcha.

Reply:
Addressed [abeb515](https://github.com/stellar/stellar-disbursement-platform-backend/pull/368/commits/abeb515c89402d855ecb6723f3e4f12c94cdd3c7)